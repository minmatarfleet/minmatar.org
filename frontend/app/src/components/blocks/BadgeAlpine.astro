---
interface Props {
    size?:  number,
}

const {
    size = 64,
} = Astro.props

import Flexblock from '@components/compositions/Flexblock.astro';
import FixedFluid from '@components/compositions/FixedFluid.astro';
---
<FixedFluid
    class="[ items-center w-full ]"
    width={`${size}px`}
    {...Astro.props}
>
    <picture x-bind:class="typeof border != typeof undefined ? 'border' : ''">
        <img x-bind:src="image" width={size} height={size} />
    </picture>
    <Flexblock gap='0'>
        <span
            x-bind:class="highlight ? 'highlight' : ''"
            x-text="title"
        />
        <template x-if="typeof subtitle != typeof undefined">
            <small x-text="subtitle" />
        </template>
    </Flexblock>
</FixedFluid>

<style lang="scss">
    .highlight {
        color: var(--highlight);
    }

    span {
        text-wrap: balance;
    }
</style>