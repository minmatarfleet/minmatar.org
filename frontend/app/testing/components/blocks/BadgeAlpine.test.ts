import { experimental_AstroContainer as AstroContainer } from "astro/container";
import { expect, test } from "vitest";
import BadgeAlpine from "@components/blocks/BadgeAlpine.astro";

test("BadgeAlpine defaults", async () => {
  const container = await AstroContainer.create();
  const result = await container.renderToString(BadgeAlpine, {});

  expect(result).toMatchSnapshot();
});
