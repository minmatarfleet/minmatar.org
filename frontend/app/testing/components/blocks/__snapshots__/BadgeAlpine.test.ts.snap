// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`BadgeAlpine defaults 1`] = `"<div class="fixed-fluid [ items-center w-full ]" data-astro-cid-ngcwkgb6="true" data-mod data-astro-cid-34go4nvk style="--fixed_fluid_gap: var(--space-s);--fixed_fluid_width: 64px;--fixed_fluid_breakpoint: 50%;">  <picture x-bind:class="typeof border != typeof undefined ? 'border' : ''" data-astro-cid-ngcwkgb6> <img x-bind:src="image" width="64" height="64" data-astro-cid-ngcwkgb6> </picture> <div class="flex-block gap-0" data-astro-cid-ngcwkgb6="true" data-mod style="--flex_block_gap: 0;">  <span x-bind:class="highlight ? 'highlight' : ''" x-text="title" data-astro-cid-ngcwkgb6></span> <template x-if="typeof subtitle != typeof undefined" data-astro-cid-ngcwkgb6> <small x-text="subtitle" data-astro-cid-ngcwkgb6></small> </template>  </div>   </div>   "`;
